lib/wcgi-runner/src/builder.rs

```diff
@@ -0,0 +1,252 @@
+/// let webc = std::fs::read("path/to/server.webc")?;
+/// let runner = Runner::builder().build_webc(webc)?;
+/// # Ok(())
+/// # }
+#[derive(Default)]
+pub struct Builder {
+    args: Vec<String>,
+    program: Option<String>,
+    dialect: Option<CgiDialect>,
+    engine: Option<Engine>,
+    env: HashMap<String, String>,
+    forward_host_env: bool,
+    mapped_dirs: Vec<(String, PathBuf)>,
+    tokio_handle: Option<Handle>,
+}
+
+impl Builder {
+    pub fn new() -> Self {
+        Builder::default()
+    }
+
+    /// Set the name of the program.
+    pub fn program(self, program: impl Into<String>) -> Self {
+        Builder {
+            program: Some(program.into()),
+            ..self
+        }
+    }
+
+    /// Add an argument to the WASI executable's command-line arguments.
+    pub fn arg(mut self, arg: impl Into<String>) -> Self {
+        self.args.push(arg.into());
+        self
+    }
+
+    /// Add multiple arguments to the WASI executable's command-line arguments.
+    pub fn args<A, S>(mut self, args: A) -> Self
+    where
+        A: IntoIterator<Item = S>,
+        S: Into<String>,
+    {
+        self.args.extend(args.into_iter().map(|s| s.into()));
+        self
+    }
+
+    /// Expose an environment variable to the guest.
+    pub fn env(mut self, name: impl Into<String>, value: impl Into<String>) -> Self {
+        self.env.insert(name.into(), value.into());
+        self
+    }
+
+    /// Expose multiple environment variables to the guest.
+    pub fn envs<I, K, V>(mut self, variables: I) -> Self
+    where
+        I: IntoIterator<Item = (K, V)>,
+        K: Into<String>,
+        V: Into<String>,
+    {
+        self.env
+            .extend(variables.into_iter().map(|(k, v)| (k.into(), v.into())));
+        self
+    }
+
+    /// Forward all of the host's environment variables to the guest.
+    pub fn forward_host_env(self) -> Self {
+        Builder {
+            forward_host_env: true,
+            ..self
+        }
+    }
+
+    /// Override the CGI dialect.
+    pub fn cgi_dialect(self, dialect: CgiDialect) -> Self {
+        Builder {
+            dialect: Some(dialect),
+            ..self
+        }
+    }
+
+    /// Map `guest_dir` to a directory on the host.
+    pub fn map_dir(mut self, guest_dir: impl Into<String>, host_dir: impl Into<PathBuf>) -> Self {
+        self.mapped_dirs.push((guest_dir.into(), host_dir.into()));
+        self
+    }
+
+    /// Map one or more `guest_dir` to a directory on the host.
+    pub fn map_dirs<I, G, H>(mut self, mapping: I) -> Self
+    where
+        I: IntoIterator<Item = (G, H)>,
+        G: Into<String>,
+        H: Into<PathBuf>,
+    {
+        for (guest_dir, host_dir) in mapping {
+            self.mapped_dirs.push((guest_dir.into(), host_dir.into()));
+        }
+
+        self
+    }
+
+    /// Pass in a [`Handle`] to a custom tokio runtime.
+    pub fn tokio_handle(self, handle: Handle) -> Self {
```

Comment:
Would be preferable to remain runtime agnostic and use the `VirtualTaskManager` trait.